common/types/try_mutex.go

```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	"sync"
+	"sync/atomic"
+	"unsafe"
+)
+
+const mutexLocked = 1 << iota
+
+// Mutex is simple sync.Mutex + ability to try to Lock.
+type TryMutex struct {
```

Comment:
Why not just use the package https://godoc.org/github.com/LK4D4/trylock
same implementation

Reply:
i did something a little more minimal then them don't want to be dependant on some small unknown repo